### PR TITLE
IO-modules

### PR DESCRIPTION
**Pull request by qualmluc**
as i mentioned, you now can use .cnf files (dimacs format) with our procedure. 

for the use in command line you can find more in my readme on my branch.

### CODE REVIEW

File: src/input/dimacs.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            convertedLine = line.strip().split(" ")
+
+            linePrefix = convertedLine[0]
+
+            # a p-line catch
+            if linePrefix == "p":
+                numberOfVariables = convertedLine[2]
+                numberOfClauses = convertedLine[3]
+
+                if convertedLine[1] == "KNF":
```
Comment (melanie-taprogge) [inline src/input/dimacs.py]: Hier müssen wir vermutlich noch mehr Bezeichnungen neben "KNF" zulassen, z.B. bei den Benchmark Problems der University of British Columbia ist die Bezeichnung "cnf"

**Comment by qualmluc** [comment]
sollte nun pullbar sein :D

File: src/input/dimacs.py (hunk 1 above)
Comment (qualmluc) [reply src/input/dimacs.py]: everything fixed, so no problem anymore!

**Comment by melanie-taprogge** [comment]
The generated CNFs in format list(np.array) contains empty arrays at the end (i.e. empty clauses), for instance:
The original CNF is satisfiable and can therefore not contain empty clauses.

[array([  4, -18,  19]), array([ 3, 18, -5]), array([ -5,  -8, -15]), array([-20,   7, -16]), array([ 10, -13,  -7]), array([-12,  -9,  17]), array([17, 19,  5]), array([-16,   9,  15]), array([ 11,  -5, -14]), array([ 18, -10,  13]), array([-3, 11, 12]), array([ -6, -17,  -8]), array([-18,  14,   1]), array([-19, -15,  10]), array([ 12,  18, -19]), array([-8,  4,  7]), array([-8, -9,  4]), array([  7,  17, -15]), array([ 12,  -7, -14]), array([-10, -11,   8]), array([  2, -15, -11]), array([9, 6, 1]), array([-11,  20, -17]), array([  9, -15,  13]), array([ 12,  -7, -17]), array([-18,  -2,  20]), array([20, 12,  4]), array([19, 11, 14]), array([-16,  18,  -4]), array([ -1, -17, -19]), array([-13,  15,  10]), array([-12, -14, -13]), array([ 12, -14,  -7]), array([-7, 16, 10]), array([ 6, 10,  7]), array([ 20,  14, -16]), array([-19,  17,  11]), array([ -7,   1, -20]), array([-5, 12, 15]), array([ -4,  -9, -13]), array([ 12, -11,  -7]), array([-5, 19, -8]), array([ 1, 16, 17]), array([ 20, -14, -15]), array([13, -4, 10]), array([14,  7, 10]), array([-5,  9, 20]), array([ 10,   1, -19]), array([-16, -15,  -1]), array([ 16,   3, -11]), array([-15, -10,   4]), array([  4, -15,  -3]), array([-10, -16,  11]), array([-8, 12, -5]), array([14, -6, 12]), array([ 1,  6, 11]), array([-13,  -5,  -1]), array([-7, -2, 12]), array([  1, -20,  19]), array([ -2, -13,  -8]), array([15, 18,  4]), array([-11,  14,   9]), array([ -6, -15,  -2]), array([  5, -12, -15]), array([-6, 17,  5]), array([-13,   5, -19]), array([20, -1, 14]), array([  9, -17,  15]), array([ -5,  19, -18]), array([-12,   8, -10]), array([-18,  14,  -4]), array([15, -9, 13]), array([ 9, -5, -1]), array([ 10, -19, -14]), array([20,  9,  4]), array([-9, -2, 19]), array([ -5,  13, -17]), array([  2, -10, -18]), array([-18,   3,  11]), array([ 7, -9, 17]), array([-15,  -6,  -3]), array([ -2,   3, -13]), array([12,  3, -2]), array([-2, -3, 17]), array([ 20, -15, -16]), array([ -5, -17, -19]), array([-20, -18,  11]), array([-9,  1, -5]), array([-19,   9,  17]), array([12, -2, 17]), array([  4, -16,  -5]), array([], dtype=float64), array([], dtype=float64), array([], dtype=float64)]

(output generated with appended benchmark file, had to change file type to .txt since .cnf isn't supported)
[uf20-01.txt](https://github.com/Luc4sQ/JML-SAT/files/11444709/uf20-01.txt)

**Review by JWittkat — APPROVED** [review]

**Review by melanie-taprogge — APPROVED** [review]
Looks great to me!

**Pull request merged**